source/loader/layers/sanitizer/asan_options.cpp

```diff
@@ -0,0 +1,126 @@
+    if (!OptionsEnvMap.has_value()) {
+        return;
+    }
+
+    const char *TrueStrings[] = {"1", "true"};
+    const char *FalseStrings[] = {"0", "false"};
+
+    auto InplaceToLower = [](std::string &S) {
+        std::transform(S.begin(), S.end(), S.begin(),
+                       [](unsigned char C) { return std::tolower(C); });
+    };
+    auto IsTrue = [&](const std::string &S) {
+        return std::any_of(std::begin(TrueStrings), std::end(TrueStrings),
+                           [&](const char *CS) { return S == CS; });
+    };
+    auto IsFalse = [&](const std::string &S) {
+        return std::any_of(std::begin(FalseStrings), std::end(FalseStrings),
+                           [&](const char *CS) { return S == CS; });
+    };
+
+    auto SetBoolOption = [&](const std::string &Name, bool &Opt) {
+        auto KV = OptionsEnvMap->find(Name);
+        if (KV != OptionsEnvMap->end()) {
+            auto Value = KV->second.front();
+            InplaceToLower(Value);
+            if (IsTrue(Value)) {
+                Opt = true;
+            } else if (IsFalse(Value)) {
+                Opt = false;
+            } else {
+                std::stringstream SS;
+                SS << "<SANITIZER>[ERROR]: \"" << Name << "\" is set to \""
+                   << Value << "\", which is not an valid setting. ";
+                SS << "Acceptable input are: for enable, use:";
+                for (auto &S : TrueStrings) {
+                    SS << " \"" << S << "\"";
+                }
+                SS << "; ";
+                SS << "for disable, use:";
+                for (auto &S : FalseStrings) {
+                    SS << " \"" << S << "\"";
+                }
+                SS << ".";
+                die(SS.str().c_str());
+            }
+        }
+    };
+
+    SetBoolOption("debug", Debug);
+    SetBoolOption("detect_locals", DetectLocals);
+    SetBoolOption("detect_privates", DetectPrivates);
+    SetBoolOption("print_stats", PrintStats);
+    SetBoolOption("detect_kernel_arguments", DetectKernelArguments);
+
+    auto KV = OptionsEnvMap->find("quarantine_size_mb");
+    if (KV != OptionsEnvMap->end()) {
+        const auto &Value = KV->second.front();
+        try {
+            auto temp_long = std::stoul(Value);
+            if (temp_long > UINT32_MAX) {
+                throw std::out_of_range("");
+            }
+            MaxQuarantineSizeMB = temp_long;
+        } catch (...) {
+            die("<SANITIZER>[ERROR]: \"quarantine_size_mb\" should be "
+                "an positive integer that smaller than or equal to "
+                "4294967295.");
+        }
+    }
+
+    KV = OptionsEnvMap->find("redzone");
+    if (KV != OptionsEnvMap->end()) {
+        const auto &Value = KV->second.front();
+        try {
+            MinRZSize = std::stoul(Value);
+            if (MinRZSize < 16) {
+                MinRZSize = 16;
+                getContext()->logger.warning("Trying to set redzone size to a "
```

Comment:
When AsanOptions is constructing, the getContext will refer to a also-constructing Context class. I think that is bad.

Reply:
offline sync with yingcong, we needn't to worry this.